src/components/List.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ListItem from './ListItem';
+import Button from './Button';
+
+const List = ({
+    tab,
+    classListName,
+    classListItemName,
+    classButtonName,
+    buttonHandler,
+    buttonText,
+    children,
+    ...rest,
+}) => (
+    <ul className={classListName}>
+      {tab.map((text, id) => (
+        <ListItem
+            key={id}
+            text={text}
+            className={classListItemName}
+            {...rest}
+        >
+            {children}
+            {buttonHandler && <Button className={classButtonName} onClickHandler={() => buttonHandler(id)} text={buttonText} />}
+        </ListItem>
+        )
+      )}
+    </ul>
+    );
+
+List.propTypes = {
+    tab: PropTypes.array.isRequired,
+    classListName: PropTypes.string,
+    classListItemName: PropTypes.string,
+    classButtonName: PropTypes.string,
+    buttonHandler: PropTypes.func,
+    buttonText: PropTypes.string,
+    children: PropTypes.node,
+}
+List.defaultProps = {
```

Comment:
if yuo are not requiring some props, you need to pass default value for them (sometimes it can be empty string `''` or `null`)